singularity-eos/eos/get_sg_eos.cpp

```diff
@@ -0,0 +1,272 @@
+//------------------------------------------------------------------------------
+// © 2021-2023. Triad National Security, LLC. All rights reserved.  This
+// program was produced under U.S. Government contract 89233218CNA000001
+// for Los Alamos National Laboratory (LANL), which is operated by Triad
+// National Security, LLC for the U.S.  Department of Energy/National
+// Nuclear Security Administration. All rights in the program are
+// reserved by Triad National Security, LLC, and the U.S. Department of
+// Energy/National Nuclear Security Administration. The Government is
+// granted for itself and others acting on its behalf a nonexclusive,
+// paid-up, irrevocable worldwide license in this material to reproduce,
+// prepare derivative works, distribute copies to the public, perform
+// publicly and display publicly, and to permit others to do so.
+//------------------------------------------------------------------------------
+
+#include <map>
+#include <ports-of-call/portability.hpp>
+#include <singularity-eos/closure/mixed_cell_models.hpp>
+#include <singularity-eos/eos/eos.hpp>
+#include <singularity-eos/eos/get_sg_eos.hpp>
+#include <singularity-eos/eos/get_sg_eos_lambdas.hpp>
+#include <singularity-eos/eos/singularity_eos.hpp>
+
+using namespace singularity;
+
+// mapping from EAP integer to
+static const std::map<const int, const unsigned long> EAPInputToBD = {
+    {-3, thermalqs::temperature | thermalqs::density},
+    {-2, thermalqs::density | thermalqs::pressure},
+    {-1, thermalqs::pressure | thermalqs::temperature},
+    {0, thermalqs::specific_internal_energy | thermalqs::density},
+    {1, thermalqs::specific_internal_energy | thermalqs::density},
+};
+
+// EAP centric arguments and function signature
+int get_sg_eos( // sizing information
```

Comment:
Is the goal for this to be moved over to EAP at some point then?

Reply:
yes, but not yet.